.github/workflows/maven-publish.yml

```diff
@@ -0,0 +1,48 @@
+# This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#apache-maven-with-a-settings-path
+
+name: Maven Publish
+
+on:
+  push: 
+    branches: [ master ]
```

Comment:
Let's rename the main branch to main to be consistent with git's new defaults?

Reply:
Do you mean renaming from "master" to "main"? Fine

I think I noticed something wrong here, on PRs we don't want to publish any packages, just run the validation checks.

Reply:
yup!

Reply:
Added a conditional for publishing a package only on the "push" event.
I'm going to rename the branch after merging this one.